src/libertem/udf/masks.py

```diff
@@ -1,11 +1,99 @@
 from libertem.common.math import prod
 import numpy as np
 
-from libertem.udf import UDF
+from libertem.udf import UDF, UDFMeta
 from libertem.common.container import MaskContainer
 from libertem.common.numba import rmatmul
 
 
+class ApplyMasksEngine:
+    def __init__(self, masks: MaskContainer, meta: UDFMeta, use_torch: bool):
+        self.masks = masks
+        self.meta = meta
```

Comment:
Just a comment, nothing to fix right now: taking a `UDFMeta` instance here means this class is right now only usable in UDFs. I do really love that this code is now reusable, and in the future we might want to lift it up further to make it usable stand-alone (possibly by extracting the relevant parameters into our own parameter class, which is then easily constructable from ` UDFMeta`.